src/main/java/space/morphanone/webizen/events/BasicRequestScriptEvent.java

```diff
@@ -178,7 +176,24 @@ public ObjectTag getContext(String name) {
             return new ElementTag(httpExchange.getRemoteAddress().toString());
         }
         else if (name.equals("query")) {
-            return new ElementTag(httpExchange.getRequestURI().getQuery());
+            String query = httpExchange.getRequestURI().getQuery();
+            return new ElementTag(query != null ? query : "");
+        }
+        else if (name.equals("query_map")) {
+            MapTag mappedValues = new MapTag();
+            String query = httpExchange.getRequestURI().getQuery();
+            try {
+                if (query != null) {
+                    for (String value : CoreUtilities.split(query, '&')) {
+                        List<String> split = CoreUtilities.split(value, '=');
+                        mappedValues.map.put(new StringHolder(split.get(0)), new ElementTag(java.net.URLDecoder.decode(split.get(1), StandardCharsets.UTF_8.name())));
```

Comment:
Both parts must be decoded, not just the second part, as I said on Discord.

Reply:
Also, `StandardCharsets.UTF_8.name()` seems like just an overcomplicated way of writing `"UTF-8"` isn't it?